datafusion/core/src/datasource/file_format/file_compression_type.rs

```diff
@@ -152,7 +153,7 @@ impl FileCompressionType {
     /// according to this `FileCompressionType`.
     pub fn convert_async_writer(
         &self,
-        w: Box<dyn AsyncWrite + Send + Unpin>,
+        w: BufWriter,
```

Comment:
I'm thinking whether it's OK to change the param type here cause it's public, but keeping `Box<dyn AsyncWrite + Send + Unpin>` makes the type incompatible.  🤔

Reply:
I think it depends on if this code is always used with `object_store` (aka if DataFusion code always writes output using the `object_store` API). If this is the case, then switching to `BufWriter` here makes sense to me

BTW I think we need to update the comments on this function to match the new implementation